### PR TITLE
Mark read on Scroll

### PR DESCRIPTION
**Pull request by mikedg1**
Description:

Implemented mark read on scroll. 

Changes Made:
Created a view modifier that allows us to track when a view scrolls upwards off screen for the first time

Added setting using @default to control this

Design decisions for discussion:

This will not hide posts immediately after scrolilng off the screen, that will only happen after a refresh or subreddit change. This decision prevents the post feed from jumping around.

In order to reduce code duplication, I simplified nested if statements that resulted in the same view being shown. This can result in less chance of the views diverging in the future, at the risk of a slightly less readable if statement. I'd love your thoughts.

I'm not too experienced in Swift, so I left out texts that would tests the logic in the modifier and view. Does it make sense to put the effort to test those?

Screenshots:
NONE

### CODE REVIEW

File: OpenArtemis/Views/Favorites Drawer/SubredditDrawerView.swift [hunk 1]
```diff
@@ -293,7 +293,7 @@ struct SubredditDrawerView: View {
     private func getSubredditFeedView() -> SubredditFeedView {
         switch defaultLaunchFeed {
         case "all":
-            return SubredditFeedView(subredditName: defaultLaunchFeed.capitalized, titleOverride: nil, appTheme: appTheme, textSizePreference: textSizePreference)
```
Comment (mikedg1) [inline OpenArtemis/Views/Favorites Drawer/SubredditDrawerView.swift:296]: TODO @mikedg1 remove change

**Comment by ejbills** [comment]
Initial review: it looks good! I’ll conduct further testing and merge the changes if the logic works well.

Furthermore, I appreciate all your efforts on the app. I’ll release your changes to the app store soon. If you have significant changes in progress that you’d like to include in the next release, please let me know, and I’ll wait until after to release the updated version. Thanks again!

**Comment by mikedg1** [comment]
> Initial review: it looks good! I’ll conduct further testing and merge the changes if the logic works well.
> 
> Furthermore, I appreciate all your efforts on the app. I’ll release your changes to the app store soon. If you have significant changes in progress that you’d like to include in the next release, please let me know, and I’ll wait until after to release the updated version. Thanks again!

No problem, I don't think there was anything else in my plans that I'm dieing to get in the next version.  I appreciate that you took my PRs!

Although, I am working on a fix for https://github.com/ejbills/OpenArtemis/issues/56
If you could just verify that it's actually a problem and I'm not just doing something wrong, that would be awesome.

Also, for some reason, I've run into a few problems with my local builds that, even before any of the submitted changes, that  have driven me mad, that don't seem to be in the App Store version. 
I tried 2 different machines, both M1's though. Both issues continued.

1- Thread 1: EXC_BAD_ACCESS (code=1, address=0x391a3a524d10) when clicking on Image media for the second time on an M1 running MacOS. Like I said, your app store build seemed great. But if I click the image of a post that eventually leads to the LazyPager(i think), the first time it shows fine. The second time, whether a different image or the same, crashes with the EXC_BAD_ACCESS error. Works fine on iPhone, iPad. I did try updating that library to the latest version with no luck.

2- Image(uiImage: UIImage(named: "AppIcon")!)
The UIImage call is always null for me, which causes a crash in Settings or during onboarding. I tried 2 different machines with the same issue. I'm using XCode 16.0. 
I think I had narrowed it down to just being AppIcons in the assets. If I change that to an image set, it loads fine even using the same images. It just doesn't make sense to me :/

I didn't post either as bugs since I'm pretty sure it's just something with my dev environment, but if you have any thoughts or think they should be posted as bugs, let me know and I'm happy to submit.

**Comment by ejbills** [comment]
In regards to point 2, ios 18 deprecated how the app was loading images. I fixed that in the latest version of main.

In regards to issue #56 - that is definitely an issue and I am able to reproduce it on my end. I was also able to reproduce the crash on the macOS version of the app, when opening a gallery of images. This will need to be fixed before I push the next app store version, as this appears to have been introduced by some macOS sequoia changes since it didnt happen previously.

**Pull request merged**